server.go

```diff
@@ -3207,6 +3207,18 @@ func (s *server) OpenChannel(
 	}
 	s.mu.RUnlock()
 
+	// We'll wait until the peer is active before beginning the channel
+	// opening process.
+	select {
+	case <-peer.activeSignal:
```

Comment:
we should avoid adding more places where we access internal members of the peer, but this works for now :) we can defer that to when we move the peer into its own package